MON-4414: Optional Monitoring Capability

rexagod opened the pull request:
Goes over the details for the `OptionalMonitoring` capability, which targets putting the in-cluster monitoring stack in a telemetry-only state. Note that the metric targets are not modified under this capability itself, but only when the telemetry collection profile is enabled.

rexagod committed:
MON-4414: OEP for optional monitoring

Goes over the details for the `OptionalMonitoring` capability, which
targets putting the in-cluster monitoring stack in a telemetry-only
state. Note that the metric targets are not modified under this
capability itself, but only when the telemetry collection profile is
enabled.

rexagod committed:
fixup! MON-4414: OEP for optional monitoring

rexagod committed:
fixup! fixup! MON-4414: OEP for optional monitoring

simonpasquier commented on enhancements/monitoring/optional-monitoring-capability.md:
```diff
@@ -0,0 +1,967 @@
+life-cycled, monitored and remediated at scale.
+-->
+
+> As a cluster administrator, I want to be able to disable as much
```
(nit)
```suggestion
> As a cluster administrator, I want to disable as much
```

simonpasquier commented on enhancements/monitoring/optional-monitoring-capability.md:
```diff
@@ -0,0 +1,967 @@
+  (implicitly enabled).
+* Ensure that disabling the capability does not affect telemetry
+  operations.
+* Owing to the reduced volume of monitoring data collected, move
```
(nit) is this a goal or a consequence?

simonpasquier commented on enhancements/monitoring/optional-monitoring-capability.md:
```diff
@@ -0,0 +1,967 @@
+  when the capability is disabled. For example, metrics-server will
+  remain enabled to ensure that the autoscaling pipelines are not
+  affected.
+* Monitoring components that other capabilities rely on, and disabling
```
I'm not sure about this point. Do we say that monitoring-plugin will be deployed?

simonpasquier commented on enhancements/monitoring/optional-monitoring-capability.md:
```diff
@@ -0,0 +1,967 @@
+
+1. Opt `Console` capability dashboards housed under CMO into
+`OptionalMonitoring` as well?
+    > No, since exporters are not disabled, and the targets are left
```
Hmm there will no platform metrics so dashboards will be empty?

simonpasquier commented on enhancements/monitoring/optional-monitoring-capability.md:
```diff
@@ -0,0 +1,967 @@
+    exception, and will remain enabled to ensure that the autoscaling
+    pipelines are not affected.
+
+6. Should the capability be named `OptionalMonitoring` or just `Monitoring`?
```
I'm still not convinced by OptionalMonitoring. What about PlatformMonitoring?

simonpasquier reviewed (commented):
Good start! Let's focus on the overview and external picture first before going to the implementation details :)

rexagod committed:
fixup! fixup! fixup! MON-4414: OEP for optional monitoring

rexagod committed:
OCPBUGS-X: Add `Telemetry` to CP set

Add `Telemetry` collection profile to the set of offered collection
profiles in CMO.

simonpasquier commented on enhancements/monitoring/optional-monitoring-capability.md:
```diff
@@ -0,0 +1,994 @@
+DPUs exist specifically to offload infrastructure overhead from host x86
+servers, achieving up to 70% CPU savings on the host. Running full monitoring
+stacks on the DPU ARM cores would defeat this purpose by consuming the limited
+resources meant for high-performance networking and infrastructure services.
```
"defeat" is a strong statement IMHO. From our conversations, I have the feeling that the current architecture has limited resources for workloads other than pure networking functions which drives a need for limiting the resource usage of platform components (especially with hosted control planes).

The consequence for this proposal is that when both RH telemetry *and* the capability are disabled, CMO shouldn't deploy Prometheus at all.

simonpasquier commented:
/reopen

rexagod commented:
/reopen

rexagod commented:
I'll revisit this once the telemetry profile effort is complete.
